src/main/java/de/kekru/struktogrammeditor/control/Controlling.java

```diff
@@ -340,12 +340,28 @@ public void actionPerformed(ActionEvent e) {
 		case struktogrammbeschreibungHinzufuegen:
 			addStruktogrammbeschriftung();
 			break;
-
+		case drucken:
+			//TODO: Implementing
+			
+			PrinterJob p = PrinterJob.getPrinterJob();
+			BufferedImage img = gibAktuellesStruktogramm().generateImage(true);
+			ImagePrintable printable = new ImagePrintable(p.defaultPage(), img);
+			
+			p.setPrintable(printable);
+			if (p.printDialog()) {
+				try {
+					p.print();
+				} catch (PrinterException e1) {
+					e1.printStackTrace();

```

Comment:
Cool idea :) 
Did not know, that it is that easy to add printing

Please add an error message in a JOptionPane dialog with the e1.getMessage(), additionally to the printStackTrace()


And you could move the code of this case-Block in a separate Method, to keep the switch-Block smaller